model/task_annotations/db.go

```diff
@@ -61,6 +69,43 @@ func (annotation *TaskAnnotation) Update() error {
 	return db.UpdateId(Collection, annotation.Id, annotation)
 }
 
+// Update updates one task_annotation.
+func (annotation *TaskAnnotation) UpdateAPIAnnotation(id string, execution int, a Annotation, m *birch.Document) error {
+	return errors.WithStack(db.Update(
+		Collection,
+		bson.M{
+			TaskIdKey:        id,
+			TaskExecutionKey: execution,
+		},
+		bson.M{
+			"$set": bson.M{
+				APIAnnotationKey: a,
+				MetadataKey:      m,
+			},
+		},
+	))
+}
+
+// Update updates one task_annotation.
+func (annotation *TaskAnnotation) UpdateUserAnnotation(id string, execution int, a Annotation, author string) error {
+	var userAnnotation = a
```

Comment:
sorry I didn't catch this the first time: why are you creating a separate userAnnotation field? you then modify the original annotation A so the source isn't getting set correctly. It'd fine just to modify a and then store it.